jwst/linearity/tests/test_linearity.py

```diff
@@ -0,0 +1,596 @@
+"""Unit tests for the non-linearity correction.
+
+Authors:
+    M. Cracraft
+"""
+from jwst.linearity import LinearityStep
+from jwst.linearity.linearity import do_correction as lincorr
+from jwst.datamodels import dqflags, LinearityModel, MIRIRampModel
+from astropy.io import fits
+import numpy as np
+
+
+def test_coeff_dq():
+    """test linearity algorithm with random data ramp (does algorithm match expected algorithm)
+    also test a variety of dq flags and expected output """
+
+    # size of integration
+    nints = 1
+    ngroups = 160
+    xsize = 1032
+    ysize = 1024
+
+    # create the data and groupdq arrays
+    # create a JWST datamodel for MIRI data
+    im = make_rampmodel(nints, ngroups, ysize, xsize)
+    im.meta.instrument.detector = 'MIRIMAGE'
+
+    # Create reference file
+    numcoeffs = 5
+
+    ref_model = LinearityModel()
+    ref_model.data = np.zeros(shape=(numcoeffs, ysize, xsize), dtype=np.float32)
+    ref_model.dq = np.zeros((ysize, xsize), dtype=int)
+
+    ref_model.meta.instrument.name = 'MIRI'
+    ref_model.meta.instrument.detector = 'MIRIMAGE'
+    ref_model.meta.subarray.xstart = 1
+    ref_model.meta.subarray.ystart = 1
+    ref_model.meta.subarray.xsize = xsize
+    ref_model.meta.subarray.ysize = ysize
+
+    # Set coefficient values in reference file to check the algorithm
+    # Equation is DNcorr = L0 + L1*DN(i) + L2*DN(i)^2 + L3*DN(i)^3 + L4*DN(i)^4
+    # DN(i) = signal in pixel, Ln = coefficient from ref file
+    # L0 = 0 for all pixels for CDP6
+    L0 = 0
+    L1 = 0.85
+    L2 = 4.62E-6
+    L3 = -6.16E-11
+    L4 = 7.23E-16
+
+    coeffs = np.asfarray([0.0e+00,  0.85, 4.62e-06,  -6.16e-11, 7.23e-16])
+
+    ref_model.data[:, 300, 500] = coeffs
+    print(ref_model.data[:, 300, 500])
```

Comment:
Probably don't want to leave a print statement in here for the CI test runs.

Reply:
agreed

Reply:
Removed in 095ff52